python/cuml/test/test_pickle.py

```diff
@@ -85,10 +85,15 @@ def pickle_save_load(tmpdir, model):
         pf.close()
         pytest.fail(e)
 
+    del model
+
     with open(pickle_file, 'rb') as pf:
         cu_after_pickle_model = pickle.load(pf)
 
-    return cu_after_pickle_model
+    if del_model:
+        return None, cu_after_pickle_model
```

Comment:
It's a little confusing to return None just so the caller overwrites a variable's value with None. Is this necessary? Since you are explicitly calling `del model` in this function, won't the memory be automatically garbage collected?

Reply:
The `del` within the function doesn't delete the object from the outer scope unfortunately. We could simplify the usage pattern with a single return though:

model = ...instantiate ,train, etc...
model = pickle_save_load(model)

hmm... but that doesn't really deallocate the memory either between save and load... Ugh, maybe we'd have to do:
```
model = ... train model etc...
pickle_fname = pickle_save(model)
del model
model = pickle_load(pickle_fname)
```

wdyt? it's not as concise, but I think it will work.

Reply:
Ah, that's right- the `del` just decrements the reference counter.

I think that example will work and the intentions are made more clear.